openjdk/mmtkUpcalls.cpp

```diff
@@ -266,6 +279,16 @@ static size_t mmtk_number_of_mutators() {
   return Threads::number_of_threads();
 }
 
+static void mmtk_prepare_for_sanity_roots_scanning() {
+#if COMPILER2_OR_JVMCI
+  DerivedPointerTable::update_pointers();
```

Comment:
Normally it is required to do `clear()` before GC, and `update_pointers()` after GC. Why do we need to call these two methods in a 'reverse' order? Are we trying to restore the state of `DerivedPointerTable` to what it is before GC so we can do another stack scanning correctly?